backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -695,6 +704,53 @@ public FutureExperimentRunDAO(
                                                     hyperparams.get(builder.getId()))),
                                     executor);
 
+                            final var futureHyperparamsFromConfigBlobs =
+                                InternalFuture.completedInternalFuture(
+                                        populateConnectionsBasedOnPrivileges)
+                                    .thenCompose(
+                                        populateConnectionsBasedOnPrivileges -> {
+                                          if (populateConnectionsBasedOnPrivileges) {
+                                            return getAllowedEntitiesByResourceType(
+                                                ModelDBActionEnum.ModelDBServiceActions.READ,
+                                                ModelDBResourceEnum.ModelDBServiceResourceTypes
+                                                    .REPOSITORY);
+                                          } else {
+                                            return InternalFuture.completedInternalFuture(
+                                                new ArrayList<>());
+                                          }
+                                        },
+                                        executor)
+                                    .thenCompose(
+                                        selfAllowedRepositoryIds -> {
+                                          if (selfAllowedRepositoryIds == null
+                                              || selfAllowedRepositoryIds.isEmpty()) {
+                                            return InternalFuture.completedInternalFuture(
+                                                    new ArrayList<
+                                                        AbstractMap.SimpleEntry<
+                                                            String, KeyValue>>())
+                                                .thenApply(MapSubtypes::from, executor);
+                                          } else {
+                                            return hyperparametersFromConfigHandler
+                                                .getExperimentRunHyperparameterConfigBlobMap(
+                                                    new ArrayList<>(ids), selfAllowedRepositoryIds);
+                                          }
+                                        },
+                                        executor);
```

Comment:
This is huge. We shouldn't inline this amount of logic. Extract to the handler.

Reply:
This is already added in separate handler 
```
return hyperparametersFromConfigHandler
                                                .getExperimentRunHyperparameterConfigBlobMap(
                                                    new ArrayList<>(ids), selfAllowedRepositoryIds);
```
and above logic is not complex it just checking repository access using `getAllowedEntitiesByResourceType`  so due to getAllowedEntitiesByResourceType function i can not move that code to handler. i can create a separate function for this inline code

Reply:
extracted code in separate function and try to explain code with adding code comments